libs/model/src/policies/ContestWorker.policy.ts

```diff
@@ -1,18 +1,77 @@
 import { schemas, type Policy } from '@hicommonwealth/core';
+import { commonProtocol, models } from '@hicommonwealth/model';
 
 const inputs = {
   ThreadCreated: schemas.events.ThreadCreated,
+  ThreadUpvoted: schemas.events.ThreadUpvoted,
   CommentCreated: schemas.events.CommentCreated,
 };
 
 export const ContestWorker: Policy<typeof inputs> = () => ({
   inputs,
   body: {
     ThreadCreated: async ({ name, payload }) => {
-      console.log(name, payload.thread);
+      console.log(name, payload);
+      const { threadId, userAddress } = payload;
+
+      const contestAddress = '0x123'; // TODO: get contest address from contest metadata projection
+      const contentUrl = `https://`; // TODO: generate content URL programmatically?
+
+      const web3Client = await createWeb3Client(threadId);
+
+      const { contentId } = await commonProtocol.contestHelper.addContent(
+        web3Client,
+        contestAddress,
+        userAddress,
+        contentUrl,
+      );
+
+      // TODO: save content ID for later use in ThreadUpvoted event
+    },
+    ThreadUpvoted: async ({ name, payload }) => {
+      console.log(name, payload);
+      const { threadId, userAddress } = payload;
+
+      const contestAddress = '0x123'; // TODO: get contest address from contest metadata projection
+      const contentId = '111'; // TODO: get content ID saved from ThreadCreated
+
+      const web3Client = await createWeb3Client(threadId);
+
+      await commonProtocol.contestHelper.voteContent(
+        web3Client,
+        contestAddress,
+        userAddress,
+        contentId,
+      );
     },
     CommentCreated: async ({ name, payload }) => {
       console.log(name, payload.comment);
     },
   },
 });
+
+const createWeb3Client = async (threadId: number) => {
```

Comment:
this should be in a protocol service lib (sound like a protocol util, where the input is a generalized content URL)

Reply:
no need to load the thread if the event contains the url